aws/fake/ec2.go

```diff
@@ -0,0 +1,162 @@
+package fake
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+)
+
+const dipSplitSize = 2
+
+type Ec2MockOutputs struct {
+	DescribeSecurityGroups *ApiResponse
+	DescribeInstances      *ApiResponse
+	DescribeInstancesPages []*ApiResponse
+	DescribeSubnets        *ApiResponse
+	DescribeRouteTables    *ApiResponse
+}
+
+type MockEc2Client struct {
+	ec2iface.EC2API
+	Outputs Ec2MockOutputs
+}
+
+func (m *MockEc2Client) DescribeSecurityGroups(*ec2.DescribeSecurityGroupsInput) (*ec2.DescribeSecurityGroupsOutput, error) {
+	if out, ok := m.Outputs.DescribeSecurityGroups.response.(*ec2.DescribeSecurityGroupsOutput); ok {
+		return out, m.Outputs.DescribeSecurityGroups.err
+	}
+	return nil, m.Outputs.DescribeSecurityGroups.err
+}
+
+func (m *MockEc2Client) DescribeInstances(*ec2.DescribeInstancesInput) (*ec2.DescribeInstancesOutput, error) {
+	if out, ok := m.Outputs.DescribeInstances.response.(*ec2.DescribeInstancesOutput); ok {
+		return out, m.Outputs.DescribeInstances.err
+	}
+	return nil, m.Outputs.DescribeInstances.err
+}
+
+func (m *MockEc2Client) DescribeInstancesPages(params *ec2.DescribeInstancesInput, f func(*ec2.DescribeInstancesOutput, bool) bool) error {
+	for _, resp := range m.Outputs.DescribeInstancesPages {
+		if out, ok := resp.response.(*ec2.DescribeInstancesOutput); ok {
+			f(out, true)
+		}
+	}
+	if len(m.Outputs.DescribeInstancesPages) != 0 {
+		return m.Outputs.DescribeInstancesPages[0].err
+	}
+	return nil
+}
+
+func (m *MockEc2Client) DescribeSubnets(*ec2.DescribeSubnetsInput) (*ec2.DescribeSubnetsOutput, error) {
+	if out, ok := m.Outputs.DescribeSubnets.response.(*ec2.DescribeSubnetsOutput); ok {
+		return out, m.Outputs.DescribeSubnets.err
+	}
+	return nil, m.Outputs.DescribeSubnets.err
+}
+
+func (m *MockEc2Client) DescribeRouteTables(*ec2.DescribeRouteTablesInput) (*ec2.DescribeRouteTablesOutput, error) {
+	if out, ok := m.Outputs.DescribeRouteTables.response.(*ec2.DescribeRouteTablesOutput); ok {
+		return out, m.Outputs.DescribeRouteTables.err
+	}
+	return nil, m.Outputs.DescribeRouteTables.err
+}
+
+func MockDSGOutput(sgs map[string]string) *ec2.DescribeSecurityGroupsOutput {
```

Comment:
Please also spell out all `Mock<shortname><something>` to `Mock<longname><something>`